src/lib/support/tests/TestSpan.cpp

```diff
@@ -366,16 +366,34 @@ void PassSpanArg(const Span<T> &)
 
 TEST(TestSpan, TestConstructorTypeDeduction)
 {
-    const uint8_t nums[]      = { 1, 2 };
-    const uint8_t otherNums[] = { 1, 2, 3 };
+    const uint8_t nums[]                                   = { 1, 2 };
+    const uint8_t otherNums[]                              = { 1, 2, 3 };
+    std::array<uint8_t, 2> numsArray                       = { 1, 2 };
+    const std::array<uint8_t, 2> numsConstArray            = { 1, 2 };
+    std::array<const uint8_t, 2> constNumsArray            = { 1, 2 };
+    const std::array<const uint8_t, 2> constNumsConstArray = { 1, 2 };
 
     // These will fail to compile if type deduction is not working correctly.
     PassSpanArg(Span(nums));
     PassSpanArg(Span(nums, 1));
+    PassSpanArg(Span(numsArray));
+    PassSpanArg(Span(constNumsArray));
 
     Span a(nums);
     EXPECT_TRUE(a.data_equal(Span<const uint8_t>(otherNums, 2)));
 
     Span b(nums, 1);
     EXPECT_TRUE(b.data_equal(Span<const uint8_t>(otherNums, 1)));
+
+    Span c(numsArray);
+    EXPECT_TRUE(c.data_equal(Span<const uint8_t>(otherNums, 2)));
```

Comment:
Should we add a check to confirm we can modify the underlying array?

```suggestion
    EXPECT_TRUE(c.data_equal(Span<const uint8_t>(otherNums, 2)));
    // Check that we can modify the span's content
    c[0] = 42;
    EXPECT_EQ(numsArray[0], 42); // Verify the underlying array was modified.

```

Reply:
Will add this, and some static_asserts about types.